libmamba/tests/src/core/test_filesystem.cpp

```diff
@@ -37,6 +37,43 @@ namespace mamba
             REQUIRE(y.u8string() == u8"日本語");
         }
 
+        TEST_CASE("to_utf8")
```

Comment:
These tests only check that `to_utf8`/`from_utf8` preserve the right expected separator, right? Maybe clarify that in the names of the tests?
Also adding unicode-using tests for these functions might clarify things, maybe similar tests to `"normalized_separators_unicode"` above will work well, with non-ANSI unicode values, which will reveal issues on Windows.